Kernel/Net/TCPSocket.cpp

```diff
@@ -191,19 +195,25 @@ KResult TCPSocket::send_tcp_packet(u16 flags, const UserOrKernelBuffer* payload,
         m_sequence_number += payload_size;
     }
 
+    auto routing_decision = route_to(peer_address(), local_address(), bound_interface());
+    if (routing_decision.is_zero())
+        return EHOSTUNREACH;
+
+    if (has_mss_option) {
+        u16 mss = routing_decision.adapter->mtu() - sizeof(IPv4Packet) - sizeof(TCPPacket);
+        TCPOptionMSS mss_option { mss };
+        memcpy(buffer.data() + sizeof(TCPPacket), &mss_option, sizeof(mss_option));
```

Comment:
Should we sanity check that we don't overrun the length of the buffer?

Reply:
Good idea. Added. :)